rocketpy/rocket/rocket.py

```diff
@@ -952,16 +996,21 @@ def add_surfaces(self, surfaces, positions):
         -------
         None
         """
-        if not isinstance(surfaces, list):
-            surfaces = [surfaces]
-            positions = [positions]
-
-        for surface, position in zip(surfaces, positions):
-            if isinstance(surface, RailButtons):
-                surface.rocket_radius = surface.rocket_radius or self.radius
-                self.rail_buttons.add(surface, position)
-            else:
+        try:
+            for surface, position in zip(surfaces, positions):
+                if not isinstance(position, (Vector, tuple, list)):
+                    position = Vector([0, 0, position])
+                else:
+                    position = Vector(position)
                 self.aerodynamic_surfaces.add(surface, position)
+                self.evaluate_single_surface_cp_to_cdm(surface, position)
+        except TypeError:
+            if not isinstance(positions, (Vector, tuple, list)):
+                positions = Vector([0, 0, positions])
+            else:
+                positions = Vector(positions)
+            self.aerodynamic_surfaces.add(surfaces, positions)
+            self.evaluate_single_surface_cp_to_cdm(surfaces, positions)
 
```

Comment:
I feel this is not the only place where we will have to deal with """vectorizing""" the position argument.
Also, the code is a bit hard to understand.

I like the idea of creating a "to_vector" method in another file and importing it in here. Smth like this:


```python
from collections.abc import Iterable

def to_vector(value):
    if isinstance(value, Vector):
        return value
    elif isinstance(value, Iterable):
        return Vector(value)
    else:
        return Vector([0, 0, value])

def process_surface(self, surface, position):
    position = to_vector(position)
    self.aerodynamic_surfaces.add(surface, position)
    self.evaluate_single_surface_cp_to_cdm(surface, position)

if isinstance(surfaces, Iterable) and isinstance(positions, Iterable):
    for surface, position in zip(surfaces, positions):
        process_surface(self, surface, position)
else:
    process_surface(self, surfaces, positions)
```

Reply:
I'll add a TODO on the code for this